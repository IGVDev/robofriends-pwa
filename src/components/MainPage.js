import React, { Component } from "react";
import CardList from "../components/CardList";
import SearchBox from "../components/SearchBox";
import Scroll from "../components/Scroll";
import Header from "../components/Header";
import "./MainPage.css";

class MainPage extends Component {
  componentDidMount() {
    this.props.onRequestRobots();
  }

  render() {
    const { searchField, onSearchChange, robots, isPending } = this.props;
    const filteredRobots = robots.filter((robot) => {
      return robot.name.toLowerCase().includes(searchField.toLowerCase());
    });

    return isPending ? (
      <h1>Loading</h1>
    ) : (
      <div className="tc">
        <Header />
        <SearchBox searchChange={onSearchChange} />
        <Scroll>
          <CardList robots={filteredRobots} />
        </Scroll>
      </div>
    );
  }
}

export default MainPage;
