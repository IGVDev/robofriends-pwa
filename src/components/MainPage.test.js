import React from "react";
import MainPage from "./MainPage";
import { shallow, mount, render } from "enzyme";
import { Provider } from "react-redux";
import "../setupTests";

let wrapper;
beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: "",
    isPending: false,
  };
  wrapper = shallow(<MainPage />);
});

it("renders MainPage without crashing", () => {});

it("expect to render App component", () => {
  const mockStore = {
    robots: [],
    searchField: "",
  };
  expect(
    shallow(
      <Provider store={mockStore}>
        <MainPage />
      </Provider>
    ).toMatchSnapshot()
  );
});
