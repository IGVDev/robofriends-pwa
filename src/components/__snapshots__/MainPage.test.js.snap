// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage renders MainPage with robots 1`] = `ShallowWrapper {}`;

exports[`MainPage renders MainPage without crashing 1`] = `undefined`;
